### DISCUSSION TITLE
custom getPreviousValue() in SmartPotentiometer

### DISCUSSION THREAD

**Discussion by Nivolp1976** [tttapa/Control-Surface#587] [General]
Hi there. I was trying to initialize assets and zero all the banks of an intelligent potentiometer, which I was able to achieve by setting the potentiometer to zero and just starting the device by doing a manual walk up from bank to bank from 0 to 127, I tried to automate the action but I couldn't do it. the banks to activate. En el segundo ejemplo intentaba hacer lo mismo con random pero no lograba activar los potenciometros.  Gracias. 

`//The idea is to move all the banks together to a position like zero and that the getPreviousValue() active.
for (int i = 1; i <= 127; i++) {
          bank.select(i);
          potentiometer.update();
          potentiometer.begin();
          potentiometer.activate();
}`

`//The idea is to move the potentiometer and change the bank every second to generate a pseudo random and that the getPreviousValue() active.
if (timer){
            RandomNumber = random(1, 40);
            bank1.select(RandomNumber) ;
            potentiometer_1.update();
            potentiometer_1.begin();
            potentiometer.activate();
}`

**Comment by Nivolp1976** [comment]
I managed to initialize all smart potentiometers to zero this way. By pressing the button I move the potentiometer to any value and when I release it, all the last states remain at that value.

` if (digitalRead(button) == LOW) {
for (int i = 1; i <= 127; i++) {
          bank.select(i);
         potentiometer.update();
         potentiometer.begin();
  }
         if (digitalRead(button) == HIGH){
 for (int i = 1; i <= 127; i++) {
          bank.select(i);
          potentiometer.update();
}`

**Comment by tttapa** [comment]
I'm not sure what you're after, but the source code of https://github.com/tttapa/Control-Surface/blob/master/src/MIDI_Outputs/Bankable/Abstract/SmartMIDIFilteredAnalog.hpp is fairly straightforward, you can easily create your own class that has the behavior you want.
